### PR TITLE
CLOUD-654 avoid setting defaults in CR

### PR DESCRIPTION
**Pull request by delgod**
[CLOUD-654](https://jira.percona.com/browse/CLOUD-654) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | failed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| data-sharded | passed |
| pitr | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/ce835e7529f5dd24c64c5cacfa9868211f65d0d8
image: `perconalab/percona-server-mongodb-operator:PR-804-ce835e75`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| init-deploy | passed |
| data-sharded | passed |
| limits | passed |
| non-voting | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/ce835e7529f5dd24c64c5cacfa9868211f65d0d8
image: `perconalab/percona-server-mongodb-operator:PR-804-ce835e75`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| demand-backup-sharded | failed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/55e010a8837a01ed6b61de5b01cfd8e164ed07b2
image: `perconalab/percona-server-mongodb-operator:PR-804-55e010a8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/55e010a8837a01ed6b61de5b01cfd8e164ed07b2
image: `perconalab/percona-server-mongodb-operator:PR-804-55e010a8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/4d3f072f622b40b1526392508d94d1df26288336
image: `perconalab/percona-server-mongodb-operator:PR-804-4d3f072f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/318c56e7fd77132480a20b17df3abf32a955e25a
image: `perconalab/percona-server-mongodb-operator:PR-804-318c56e7`

File: pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go [hunk 1]
```diff
@@ -387,13 +387,7 @@ func (r *ReconcilePerconaServerMongoDBRestore) getBackup(cr *psmdbv1.PerconaServ
...
 	err := r.client.Status().Update(context.TODO(), cr)
 	if err != nil {
-		// maybe it's k8s v1.10 and earlier (e.g. oc3.9) that doesn't support status updates
-		// so try to update whole CR
-		//TODO: Update will not return error if user have no rights to update Status. Do we need to do something?
-		err := r.client.Update(context.TODO(), cr)
-		if err != nil {
-			return fmt.Errorf("send update: %v", err)
-		}
+		return fmt.Errorf("send update: %v", err)
```
Comment (egegunes) [inline pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go:390]: let's use `errors.Wrap` here too

File: pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go [hunk 2]
```diff
@@ -223,13 +223,7 @@ func (r *ReconcilePerconaServerMongoDBBackup) checkFinalizers(cr *psmdbv1.Percon
 func (r *ReconcilePerconaServerMongoDBBackup) updateStatus(cr *psmdbv1.PerconaServerMongoDBBackup) error {
 	err := r.client.Status().Update(context.TODO(), cr)
```
Comment (egegunes) [inline pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go:224]: should we use `RetryOnConflict` here too? 

https://github.com/percona/percona-server-mongodb-operator/blob/main/pkg/controller/perconaservermongodb/status.go#L216-L229

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/318c56e7fd77132480a20b17df3abf32a955e25a
image: `perconalab/percona-server-mongodb-operator:PR-804-318c56e7`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/91dac3a2478af65b9a77557d1d39cd157ace757f
image: `perconalab/percona-server-mongodb-operator:PR-804-91dac3a2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/6b755c157a5e382c95fcef912f3d29b52d8537b0
image: `perconalab/percona-server-mongodb-operator:PR-804-6b755c15`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| demand-backup-sharded | passed |
| users | passed |
| upgrade | failed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/ee0cc8a82a8f3eaf7f6f8c7e4ece15cc2023a3a1
image: `perconalab/percona-server-mongodb-operator:PR-804-ee0cc8a8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| monitoring-2-0 | passed |
| limits | passed |
| arbiter | passed |
| scaling | passed |
| service-per-pod | passed |
| security-context | passed |
| liveness | passed |
| rs-shard-migration | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/ee0cc8a82a8f3eaf7f6f8c7e4ece15cc2023a3a1
image: `perconalab/percona-server-mongodb-operator:PR-804-ee0cc8a8`

File: pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go (hunk 1 above)
Comment (delgod) [reply pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go]: fixed

File: pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go (hunk 2 above)
Comment (delgod) [reply pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/804/commits/73c82b41586c258fb0a5768215fd41149d1019bf
image: `perconalab/percona-server-mongodb-operator:PR-804-73c82b41`

**Review by egegunes — APPROVED** [review]

**Pull request merged**